dev/plugins/hu.elte.txtuml.validation/src/hu/elte/txtuml/validation/visitors/DataTypeVisitor.java

```diff
@@ -29,18 +28,15 @@ public boolean visit(FieldDeclaration node) {
 		if (!ElementTypeTeller.isFinal(node)) {
 			collector.report(new MutableDataTypeField(collector.getSourceInfo(), node));
 		}
-		if (!Utils.isBasicType(node.getType(), false)
-				&& !ElementTypeTeller.isDataType(node.getType().resolveBinding())) {
+		if (!Utils.isAllowedAttributeType(node.getType(), false)) {
 			collector.report(new InvalidDataTypeField(collector.getSourceInfo(), node));
 		}
 		return false;
 	}
 
 	@Override
 	public boolean visit(MethodDeclaration node) {
-		if (!node.isConstructor()) {
-			collector.report(new InvalidDataTypeMethod(collector.getSourceInfo(), node));
-		}
+
 		// TODO: check body
```

Comment:
According to the [Javadoc](https://github.com/ELTE-Soft/txtUML/blob/c4d09e986a5a93b6eaa86e270eb6b5482de72329/dev/plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/DataType.java#L51-L54), it is.